File: app/src/main/java/com/desuzed/clocknweather/CustomCircle.java
package com.desuzed.clocknweather;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.RectF;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.Nullable;

public class CustomCircle extends View {
    public CustomCircle(Context context) {
        this(context, null);
    }

    public CustomCircle(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public CustomCircle(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        drawCircle(canvas);
    }

    private void drawCircle (Canvas canvas){
        Paint paint = new Paint();
        paint.setColor(Color.BLUE);
        canvas.drawArc(new RectF(), 0, 300, false, paint);
        canvas.drawCircle(getPivotX(), getPivotY(), 50, paint);
    }
}

File: app/src/main/java/com/desuzed/clocknweather/util/CustomCircle.kt
package com.desuzed.clocknweather.util

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.RectF
import android.util.AttributeSet
import android.view.View

class CustomCircle @JvmOverloads constructor(
    context: Context?,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : View(context, attrs, defStyleAttr) {
    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        drawCircle(canvas)
    }

    private fun drawCircle(canvas: Canvas) {
        val paint = Paint()
        paint.color = Color.BLUE
        canvas.drawArc(RectF(), 0f, 300f, false, paint)
        canvas.drawCircle(pivotX, pivotY, 50f, paint)
    }
}